frontend/src/metabase/dashboard/components/Dashboard/Dashboard.unit.spec.tsx

```diff
@@ -0,0 +1,171 @@
+  const mockLoadDashboardParams = jest.fn();
+  const mockFetchDashboard = jest.fn(() => mockDashboard);
+  const mockFetchDashboardCardData = jest.fn();
+  const mockFetchDashboardCardMetadata = jest.fn();
+
+  function TestComponent(props: TestedProps) {
+    return (
+      <DashboardComponent
+        dashboard={mockDashboard}
+        dashboardId={props.dashboardId}
+        parameters={props.parameters}
+        parameterValues={props.parameterValues}
+        loadDashboardParams={mockLoadDashboardParams}
+        fetchDashboard={mockFetchDashboard}
+        fetchDashboardCardData={mockFetchDashboardCardData}
+        fetchDashboardCardMetadata={mockFetchDashboardCardMetadata}
+        // stuff below doesn't change
+        location={global.location}
+        isAdmin={false}
+        isFullscreen={false}
+        isNightMode={false}
+        isSharing={false}
+        isEditing={false}
+        isEditingParameter={false}
+        isNavbarOpen={false}
+        isHeaderVisible={false}
+        isAdditionalInfoVisible={false}
+        isNavigatingBackToDashboard={false}
+        editingOnLoad={false}
+        addCardOnLoad={null}
+        isAutoApplyFilters={false} // TODO true or false?
+        dashcardData={{}}
+        selectedTabId={props.selectedTabId}
+        draftParameterValues={{}}
+        editingParameter={{}}
+        sidebar={{ name: "any", props: {} }}
+        addCardToDashboard={jest.fn()}
+        addParameter={jest.fn()}
+        archiveDashboard={jest.fn()}
+        cancelFetchDashboardCardData={jest.fn()}
+        initialize={jest.fn()}
+        onRefreshPeriodChange={jest.fn()}
+        updateDashboardAndCards={jest.fn()}
+        setDashboardAttributes={jest.fn()}
+        setEditingDashboard={jest.fn()}
+        setErrorPage={jest.fn()}
+        setSharing={jest.fn()}
+        setParameterValue={jest.fn()}
+        setEditingParameter={jest.fn()}
+        setParameterIndex={jest.fn()}
+        onUpdateDashCardVisualizationSettings={jest.fn()}
+        onUpdateDashCardColumnSettings={jest.fn()}
+        onReplaceAllDashCardVisualizationSettings={jest.fn()}
+        onChangeLocation={jest.fn()}
+        onSharingClick={jest.fn()}
+        onEmbeddingClick={jest.fn()}
+        toggleSidebar={jest.fn()}
+        closeSidebar={jest.fn()}
+        closeNavbar={jest.fn()}
+      />
+    );
+  }
+
+  const { rerender } = renderWithProviders(
+    <TestComponent
+      dashboardId={props.dashboardId}
+      selectedTabId={props.selectedTabId}
+      parameters={props.parameters}
+      parameterValues={props.parameterValues}
+    />,
+  );
+
+  await waitForLoaderToBeRemoved();
+
+  return {
+    rerender: (props: TestedProps) => rerender(<TestComponent {...props} />),
+    mockLoadDashboardParams,
+    mockFetchDashboard,
+    mockFetchDashboardCardData,
+    mockFetchDashboardCardMetadata,
+  };
+}
+
+describe("Dashboard data fetching", () => {
+  const DEFAULT_PROPS: TestedProps = {
```

Comment:
We usually use these props in `setup` function so we don't need to deal with spreading

Reply:
That's actually a very good catch, thanks!